### PR TITLE
[GTB-14] Welcome message implementation

### PR DESCRIPTION
**Pull request by nour-elhabbal**
Co-authored-by: Omar Lawatey <omarlawatey@gmail.com>

Message to the team leader:
hey Omar, I did the welcome message as an embed not as an image, because i tried a lot to do it as an image but i just failed as this is very complicated process. wanna hear your review about my code asap <3

### CODE REVIEW

File: src/actionHandlers/welcome.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  rulesChannel: '1086033688871063671',
+};
+
+export const welcome = (member: GuildMember) => {
+  if (member.user.bot) {
+    member.roles.add(roles.bot);
+    return;
+  }
+
+  member.roles.member;
```
Comment (OmarLawaty) [inline src/actionHandlers/welcome.ts]: Why did you leave this line ?

File: src/actionHandlers/welcome.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  member: '1086033687109455989',
+  bot: '1086033687109455985',
+};
+
+const channels = {
+  welcomeChannel: '1086033689143676980',
+  rulesChannel: '1086033688871063671',
+};
+
+export const welcome = (member: GuildMember) => {
```
Comment (OmarLawaty) [inline src/actionHandlers/welcome.ts]: Remember to change the function name to onWelcome

File: src/actions/onGuildMemberAdd.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { GuildMember } from 'discord.js';
+import { welcome } from '../actionHandlers/welcome';
```
Comment (OmarLawaty) [inline src/actions/onGuildMemberAdd.ts]: Separate local imports from module imports using new line

**Review by OmarLawaty — CHANGES_REQUESTED** [review]
Over all good Job

File: src/actionHandlers/welcome.ts (hunk 1 above)
Comment (nour-elhabbal) [reply src/actionHandlers/welcome.ts]: just forgot to remove it :<

Comment (OmarLawaty) [reply src/actionHandlers/welcome.ts]: np :)

File: src/actionHandlers/welcome.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  member.roles.add(roles.member);
+
+  const welcomeChannel = member.guild.channels.cache.get(channels.welcomeChannel) as TextChannel;
+
+  welcomeChannel.send(`
+  >>>   \`#\` **Welcome** ${userMention(member.id)} to out server !
+  \`#\` We inform u to read our **rules** \`:\` ${channelMention(channels.rulesChannel)}
+  \`#\` Total members \`:\` **${member.guild.memberCount}**
+  \`#\` Enjoy with us :heart_on_fire:
+    `);
```
Comment (OmarLawaty) [inline src/actionHandlers/welcome.ts]: Add ".then()" after the role add

**Review by OmarLawaty — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -2,6 +2,7 @@
   "name": "galactics-bot-enhanced",
   "version": "1.0.0",
   "description": "Galactics Official bot",
+  "type": "module",
```
Comment (OmarLawaty) [inline package.json]:
```suggestion
  "type": "module",
```
This shouldn't be added

**Review by OmarLawaty — CHANGES_REQUESTED** [review]

**Review by OmarLawaty — APPROVED** [review]

**Pull request merged**